Content.Shared/Projectiles/SharedProjectileSystem.cs

```diff
@@ -169,32 +170,55 @@ private void OnAttemptPacifiedThrow(Entity<EmbeddableProjectileComponent> ent, r
     /// </summary>
     private void AfterProjectileHit(EntityUid uid, ProjectileComponent component, ref AfterProjectileHitEvent args)
     {
-        if (!TryComp<CanPenetrateComponent>(uid, out var damageAfterCollide))
+        if (!TryComp<CanPenetrateComponent>(uid, out var canPenetrate))
             return;
 
+        //Delete the projectile if it has no penetration power left.
+        if (canPenetrate.PenetrationPower <= 0)
+        {
+            QueueDel(uid);
+            return;
+        }
+
         //Delete the projectile if it hits an entity with a CollisionLayer that has a higher value than it's PenetrationLayer.
         //This allows a projectile to only penetrate a specific set of entities.
-        if (damageAfterCollide.PenetrationLayer != null)
+        if (canPenetrate.PenetrationLayer != null)
         {
-            if (args.Fixture.CollisionLayer > (int) damageAfterCollide.PenetrationLayer ||
-                damageAfterCollide.PenetrationPower == 0)
+            if (args.Fixture.CollisionLayer > (int) canPenetrate.PenetrationLayer)
             {
                 QueueDel(uid);
                 return;
             }
         }
 
         //Allow the projectile to deal damage again.
-        if(damageAfterCollide.DamageAfterCollide)
+        if(canPenetrate.DamageAfterCollide)
             component.DamagedEntity = false;
 
         //If the projectile has a limit on the amount of penetrations, reduce it.
-        if (damageAfterCollide.PenetrationPower != null)
-            damageAfterCollide.PenetrationPower -= 1;
+        if (canPenetrate.PenetrationPower != null)
+        {
+            //If it's CollisonLayer is higher than the MachineLayer it's a "hard" target and
+            //deducts more penetration power.
+            if (args.Fixture.CollisionLayer > (int) CollisionGroup.MachineLayer)
+                canPenetrate.PenetrationPower -= 1;
+
+            //If the target's CollisonLayer is equal to or lower than the MachineLayer,
+            //it's a "soft" target and deducts less penetration power.
+            else
+                canPenetrate.PenetrationPower -= 0.5f;
+
```

Comment:
Collision layers are a bit mask you can't just do an int check like this as it doesn't make sense, see https://github.com/space-wizards/space-station-14/issues/25550